test/gtest/common/mem_buffer.h

```diff
@@ -8,7 +8,10 @@
 #define GTEST_MEM_BUFFER_H_
 
 #include <ucs/memory/memory_type.h>
+#ifndef UINT8_MAX
+#define __STDC_LIMIT_MACROS
```

Comment:
is the problem is for C++ code compilation? so this macro should be define in C++ code or include gtest/common/test.h which defines it

Reply:
But this is tests - so it's a C++ code. Including "test.h" in mem_buffer.h fails to compile:
```
In file included from ./common/mem_buffer.h:11:0,
                 from common/test_helpers.h:13,
                 from common/main.cc:14:
./common/test.h:125:5: error: 'ptr_vector' does not name a type
     ptr_vector<scoped_setenv>       m_env_stack;
     ^
./common/test.h:139:24: error: 'test_skip_exception' does not name a type
     void skipped(const test_skip_exception& e);
                        ^
./common/test.h:139:45: error: ISO C++ forbids declaration of 'e' with no type [-fpermissive]
     void skipped(const test_skip_exception& e);
                                             ^
./common/test.h:193:11: error: 'ptr_vector' in namespace 'ucs' does not name a type
     const ucs::ptr_vector<T>& entities() const {
           ^
./common/test.h:219:5: error: 'ptr_vector' in namespace 'ucs' does not name a type
     ucs::ptr_vector<T> m_entities;
     ^
./common/test.h: In member function 'T& ucs::entities_storage<T>::sender()':
./common/test.h:198:17: error: 'm_entities' was not declared in this scope
         return *m_entities.front();
                 ^
./common/test.h: In member function 'T& ucs::entities_storage<T>::receiver()':
./common/test.h:202:17: error: 'm_entities' was not declared in this scope
         return *m_entities.back();
                 ^
./common/test.h: In member function 'T& ucs::entities_storage<T>::e(size_t)':
./common/test.h:206:16: error: 'm_entities' was not declared in this scope
         return m_entities.at(idx);
                ^
./common/test.h: In member function 'void ucs::entities_storage<T>::skip_loopback()':
./common/test.h:215:39: error: there are no arguments to 'UCS_TEST_SKIP_R' that depend on a template parameter, so a declaration of 'UCS_TEST_SKIP_R' must be available [-fpermissive]
             UCS_TEST_SKIP_R("loopback");
                                       ^
./common/test.h:215:39: note: (if you use '-fpermissive', G++ will accept your code, but allowing the use of an undeclared name is deprecated)
make[1]: *** [common/gtest-main.o] Error 1
```

Reply:
Circular dependency `test_helpers.h -> mem_buffer.h  -> test.h -> test_helpers .h`

Reply:
can remove `test_helpers.h -> mem_buffer.h ` include (move it to only to CC files which actually use it)